libseccomp/src/lib.rs

```diff
@@ -250,6 +250,105 @@ impl From<&ScmpArgCompare> for scmp_arg_cmp {
     }
 }
 
+#[rustfmt::skip]
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __private_scmp_cmp_arg {
+    (arg0) => { 0 };
+    (arg1) => { 1 };
+    (arg2) => { 2 };
+    (arg3) => { 3 };
+    (arg4) => { 4 };
+    (arg5) => { 5 };
+}
+
+/// A macro to create [`ScmpArgCompare`] in a more elegant way.
+///
+/// ```
+/// use libseccomp::{ScmpArgCompare, ScmpCompareOp, scmp_cmp};
+///
+/// assert_eq!(
+///     scmp_cmp!($arg0 != 123),
+///     ScmpArgCompare::new(0, ScmpCompareOp::NotEqual, 123),
+/// );
+/// assert_eq!(
+///     scmp_cmp!($arg1 < 123),
+///     ScmpArgCompare::new(1, ScmpCompareOp::Less, 123),
+/// );
+/// assert_eq!(
+///     scmp_cmp!($arg2 <= 123),
+///     ScmpArgCompare::new(2, ScmpCompareOp::LessOrEqual, 123),
+/// );
+/// assert_eq!(
+///     scmp_cmp!($arg3 == 123),
+///     ScmpArgCompare::new(3, ScmpCompareOp::Equal, 123),
+/// );
+/// assert_eq!(
+///     scmp_cmp!($arg4 >= 123),
+///     ScmpArgCompare::new(4, ScmpCompareOp::GreaterEqual, 123),
+/// );
+/// assert_eq!(
+///     scmp_cmp!($arg5 > 123),
+///     ScmpArgCompare::new(5, ScmpCompareOp::Greater, 123),
+/// );
+/// assert_eq!(
+///     scmp_cmp!($arg0 & 0x0f0 == 123),
+///     ScmpArgCompare::new(0, ScmpCompareOp::MaskedEqual(0x0f0), 123),
+/// );
+/// ```
+#[macro_export]
+macro_rules! scmp_cmp {
+    ($_:tt $arg:tt != $datum:expr) => {
+        $crate::ScmpArgCompare::new(
```

Comment:
Can we just use `crate`, not `$crate`?
currently, I think we do not have to use `$` in macro.
The same is true for other places

Reply:
No. `crate` != `$crate`.

- `crate` = the current `crate`. In case of a macro, this is the crate where the macro is expanded (e.g. `my_awesome_program`).
- `$crate` = the crate where the macro is defined (e.g. `::libseccomp`). Note however that it is not always `::libseccomp`, it can also be `::libseccomp01` for example if you rename it (https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml).

Reply:
Sorry for my misunderstanding and thank you for the information.